contracts/OwnerData.sol

```diff
@@ -0,0 +1,94 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+
+import "@openzeppelin/contracts/utils/Context.sol";
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import "@openzeppelin/contracts/utils/Strings.sol";
+
+contract FF {
+    function ownerOf(uint256 tokenId) public view returns (address) {}
+}
+
+string constant SIGNED_MESSAGE = "Feral File is requesting authorization to write your sound piece to contract";
+
+contract OwnerData is Context {
+    struct Data {
+        address owner;
+        bytes dataHash;
+        string metadata;
+    }
+
+    mapping(address => mapping(uint256 => Data[])) private _tokenData;
+
+    function add(address contractAddress, uint256 tokenID, Data calldata data) external {
+        require(_isOwner(contractAddress, tokenID, _msgSender()), "OwnerData: caller is not the owner");
+        require(data.owner == _msgSender(), "OwnerData: data owner mismatch");
+        _updateData(contractAddress, tokenID, data);
+    }
+
+    function signedAdd(
+        address contractAddress,
+        uint256 tokenID,
+        bytes memory signature,
+        Data calldata data
+    ) external {
+        address signer = _verifySignature(signature);
+        require(_isOwner(contractAddress, tokenID, signer), "OwnerData: signer is not the owner");
+        require(data.owner == signer, "OwnerData: data owner mismatch");
+        _updateData(contractAddress, tokenID, data);
+    }
+
+    function remove(address contractAddress, uint256 tokenID) external {
```

Comment:
Let's remove this `remove` function, as the new requirement is, that each owner can only record once on their chain, and it stays permanently (without being updated).